### PR TITLE
EAR 489  - Route to or skip confirmation question

### PR DESCRIPTION
**Pull request by tbaggaley**
### What is the context of this PR?

Previously Author had no means to conditionally display confirmation questions based on the answer to a previous question. This PR adds the ability to add skip conditions to confirmation questions; this allows creating the scenario used by some MBS form types - where a question confirmation is only shown when a respondent answers "0" to a previous question - within Author.

[JIRA Ticket](https://collaborate2.ons.gov.uk/jira/browse/EAR-489)
[Prototype](https://overflow.io/s/AO1N70BF?node=a327a2e6&prototype)

#### Overview of changes
- Enable `Logic` tab
- Add custom `NoRouting` page
- Add `Skippable` interface to GraphQL schema to facilitate centralising skip logic functionality between `QuestionPage` & `QuestionConfirmation`
- Update skip logic mutations to operate on `Skippable` instead of `Page`
- Ensure all existing functionality works with change to `Skippable` interface
- UI changes - can now add skip conditions to question confirmations in Author
- Add validation to question confirmation skip conditions
- Add question confirmation to validations subscription
- Move shared logic components into `App/shared` folder
- Update publisher to add skip conditions to confirmation questions

#### Author UI/API
- Open/create a questionnaire
- Add a question confirmation page to a question
- Logic tab should be available
- Routing tab available but button disabled and custom message shown as per prototype
- Skip logic page available and skip logic can be added as per its usual behaviour elsewhere
- Validation should work on the skip logic as per usual behaviour elsewhere

#### Publisher / runner
- Open the survey in runner (e.g. via "View survey")
- Fill out the survey and check to see if your skip logic was appropriately applied

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process

#### Skip logic page
Screenshot 2021-01-07 at 19 56 08

#### Routing page
Screenshot 2021-01-07 at 19 58 31

### CODE REVIEW

File: eq-author/src/App/shared/Logic/SkipLogic/SkipLogicPage/NoSkipConditions.js [hunk 1]
```diff
@@ -34,36 +34,46 @@ const Paragraph = styled.p`
...
+  ${props =>
+    props.isFirstQuestion &&
+    `
      pointer-events: none;
      opacity: 0.6;`};
 `;
 
 const SkipConditionsSetMsg = ({
-  onAddSkipCondtions,
+  onAddSkipConditions,
```
Comment (tjamesmac) [inline eq-author/src/App/shared/Logic/SkipLogic/SkipLogicPage/NoSkipConditions.js:45]: nice

File: eq-author/src/App/shared/Logic/SkipLogic/mutations.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+export const useCreateSkipCondition = ({ parentId }) =>
+  createMutationHook(CREATE_SKIP_CONDITION, { parentId });
+
+export const useDeleteSkipConditions = ({ parentId }) =>
+  createMutationHook(DELETE_SKIP_CONDITIONS, { parentId });
+
+export const useDeleteSkipCondition = ({ id }) =>
+  createMutationHook(DELETE_SKIP_CONDITION, { id });
```
Comment (tjamesmac) [inline eq-author/src/App/shared/Logic/SkipLogic/mutations.js:23]: Very nice

**Review by tjamesmac — CHANGES_REQUESTED** [review]
Okay so the only thing I've noticed is that the `NoRouting` page has no light grey border around it.

The SkipCondition page has a `<Panel>` around the container. All this does is add the border and related properties

File: eq-author/src/App/shared/Logic/SkipLogic/mutations.js (hunk 2 above)
Comment (tbaggaley) [reply eq-author/src/App/shared/Logic/SkipLogic/mutations.js:23]: Cheers Tom - updated now to include Panel.

**Review by tjamesmac — APPROVED** [review]
That's awesome Tom

**Review by jasehumphr — APPROVED** [review]
bernard-matthews-pic-pa-564541326

### Beautiful

**Pull request merged**